File: Dec2020/15Dec/InSearchOfSamosa.cpp
/*
Problem Name: In search of Samosa
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/in-search-of-samosa-pledge-easy-3/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int n, k;
    cin>>n>>k;
    int S[n];
    for(int i=0;i<n;++i)
        cin>>S[i];
    sort(S,S+n);

    int samosas=0, sum=0, i=0;
    while(i<n){
        if(sum+S[i]<=k){
            sum += S[i];
            samosas++;
            i++;
        }
        else
            break;
    }
    cout<<samosas;
    return 0;
}

File: Dec2020/15Dec/InSearchOfSamosa.java
/*
Problem Name: In search of Samosa
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/in-search-of-samosa-pledge-easy-3/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int n = in.nextInt(), k =in.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++)
            arr[i] = in.nextInt();
        Arrays.sort(arr);
        int cnt = 0;
        for(int i=0;i<n;i++){
            k -= arr[i];
            if(k < 0)
                break;
            cnt++;
        }
        out.println(cnt);
        out.flush();
    }
}
